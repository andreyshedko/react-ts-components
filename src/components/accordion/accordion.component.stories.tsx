import React from 'react';
import { AccordionComponent } from './accordion.component';

import {
    Title,
    Subtitle,
    Description,
    Primary,
    ArgsTable,
    Stories,
    PRIMARY_STORY,
} from '@storybook/addon-docs/blocks';


export default {
    component: AccordionComponent,
    title: 'Accordion Component',
    parameters: {
        docs: {
            page: () => (
                <>
                    <Title />
                    <Subtitle />
                    <Description />
                    <Primary />
                    <ArgsTable story={PRIMARY_STORY} />
                    <Stories />
                </>
            ),
        }
    },
};

export const accordionComponent = () =>
    <AccordionComponent items={[
        {
            title: 'Test Title 1',
            body: {
                title: 'Body Title 1',
                element: 'sasasasasasasass'
            },
            isOpened: false
        },
        {
            title: 'Test Title 2',
            body: {
                title: 'Body Title 2',
                element: <>React Node</>
            },
            isOpened: false
        }
    ]} />;