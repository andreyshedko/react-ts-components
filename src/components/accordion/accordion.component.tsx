import React, { PureComponent } from "react";

import { AccordionItemComponent, AccordionItem } from "../accordion-item/accordion-item.component";

export interface AccordionComponentProps {
    items: AccordionItem[];
}

export interface AccordionComponentState {
    items: AccordionItem[];
}

export class AccordionComponent extends PureComponent<AccordionComponentProps, AccordionComponentState> {

    state: Readonly<AccordionComponentState> = {
        items: []
    };

    componentDidMount() {
        this.setState({items: this.props.items});
    }

    setComponentsState(index: number) {
        // Close all but not on this index
        this.setState(prevState => ({
            items: [...prevState.items.map((item, _index) => {
                if (index === _index) {
                    item.isOpened = !item.isOpened;
                } else {
                    item.isOpened = false;
                }
                return item;
            })
        ]
        }));
    }

    render() {
        return (
            <>
                {this.state.items.map((item: AccordionItem, index) => {
                    return (
                        <>
                            <AccordionItemComponent
                                key={index}
                                isOpened={item.isOpened}
                                config={item}
                                onClick={() => {
                                    this.setComponentsState(index)
                                }} />
                        </>
                    )
                })
                }
            </>
        )
    }
};