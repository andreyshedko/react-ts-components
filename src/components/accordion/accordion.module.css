@import "../../variables.css"; 

@font-face {
    font-family: "OpenSans";
    src: url("https://fonts.googleapis.com/css?family=Open+Sans:400,700&subset=cyrillic&display=swap");
    font-display: swap;
  }

.accordion {
    font-family: "OpenSans", sans-serif;
}

.accordion .head {
    width: 100%;
    min-height: 0.5rem;
    background-color: var(--gray-3);
    display: flex;
    flex-flow: row nowrap;
    justify-content: flex-start;
    align-items: center;
    padding: var(--sm)
}

.accordion .head .icon {
    cursor: pointer;
    margin-right: calc(var(--sm) / 2 );
}

.accordion .item {
    width: 100%;
    min-height: 1rem;
    background-color: var(--gray-1);
    padding: var(--sm);
}

.accordion .hidden {
    display: none;
}

.accordion .visible {
    display: initial;
}