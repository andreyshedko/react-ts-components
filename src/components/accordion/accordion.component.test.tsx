import React from "react";

import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { AccordionComponent } from "./accordion.component";

configure({ adapter: new Adapter() });

describe("Accordion Component", () => {
    test("should display items", async () => {
        const component = renderer.create(
            <AccordionComponent
                items={
                    [
                        {
                            title: 'Test Title 1',
                            body: {
                                title: 'Body Title 1',
                                element: 'sasasasasasasass'
                            },
                            isOpened: false
                        },
                        {
                            title: 'Test Title 2',
                            body: {
                                title: 'Body Title 2',
                                element: <>React Node</>
                            },
                            isOpened: false
                        }
                    ]
                } />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});