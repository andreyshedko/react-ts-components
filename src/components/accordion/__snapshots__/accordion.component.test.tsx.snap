// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component should display items 1`] = `
Array [
  <main
    className="sc-AxjAm cwuoKc"
    tabIndex={0}
  >
    <section
      className="sc-AxirZ gjKCvg"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="sc-AxhCb kLnTaK"
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-chevron-down fa-w-14 "
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
      <h3
        className="sc-AxiKw ibCGdu"
      >
        Test Title 1
      </h3>
    </section>
    <section
      className="sc-AxhUy fJxdGm"
    >
      <h4>
        Body Title 1
      </h4>
      <div
        className="sc-AxgMl cwaAQQ"
      >
        sasasasasasasass
      </div>
    </section>
  </main>,
  <main
    className="sc-AxjAm cwuoKc"
    tabIndex={0}
  >
    <section
      className="sc-AxirZ gjKCvg"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="sc-AxhCb kLnTaK"
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-chevron-down fa-w-14 "
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
      <h3
        className="sc-AxiKw ibCGdu"
      >
        Test Title 2
      </h3>
    </section>
    <section
      className="sc-AxhUy fJxdGm"
    >
      <h4>
        Body Title 2
      </h4>
      <div
        className="sc-AxgMl cwaAQQ"
      >
        React Node
      </div>
    </section>
  </main>,
]
`;
